src/modules/hip/kernel/fmadd_scalar.hpp

```diff
@@ -0,0 +1,134 @@
+    uint dstIdx = (id_z * dstStridesCDH.y) + (id_y * dstStridesCDH.z) + id_x;
+
+    d_float8 val_f8;
+    for(int c = 0; c < channels; c++)
+    {
+        rpp_hip_load8_and_unpack_to_float8(srcPtr + srcIdx, &val_f8);
+        fmaf_scalar_hip_compute(&val_f8, &fmaddParams_f2);
+        rpp_hip_pack_float8_and_store8(dstPtr + dstIdx, &val_f8);
+        srcIdx += srcStridesCDH.x;
+        dstIdx += dstStridesCDH.x;
+    }
+}
+
+__global__ void fmadd_scalar_ndhwc_hip_tensor(float *srcPtr,
+                                              uint2 srcStridesDH,
+                                              float *dstPtr,
+                                              uint2 dstStridesDH,
+                                              float2 fmaddParams_f2,
+                                              RpptROI3DPtr roiGenericSrc)
+{
+    int id_x = (hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x) * 8;        // WC - inner most dim vectorized
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;              // H - second to inner
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;              // D - outer most dim
+
+    if ((id_z >= roiGenericSrc->xyzwhdROI.roiDepth) || (id_y >= roiGenericSrc->xyzwhdROI.roiHeight) || (id_x >= roiGenericSrc->xyzwhdROI.roiWidth))
+    {
+        return;
+    }
+
+    uint srcIdx = ((id_z + roiGenericSrc->xyzwhdROI.xyz.z) * srcStridesDH.x) + ((id_y + roiGenericSrc->xyzwhdROI.xyz.y) * srcStridesDH.y) + (id_x + roiGenericSrc->xyzwhdROI.xyz.x) * 3;
+    uint dstIdx = (id_z * dstStridesDH.x) + (id_y * dstStridesDH.y) + id_x * 3;
+
+    d_float24 val_f24;
+    rpp_hip_load24_pkd3_and_unpack_to_float24_pln3(srcPtr + srcIdx, &val_f24);
+    fmaf_scalar_hip_compute(&val_f24, &fmaddParams_f2);
+    rpp_hip_pack_float24_pln3_and_store24_pkd3(dstPtr + dstIdx, &val_f24);
+}
+
+RppStatus hip_exec_fmadd_scalar_tensor(Rpp32f *srcPtr,
+                                       RpptGenericDescPtr srcGenericDescPtr,
+                                       Rpp32f *dstPtr,
+                                       RpptGenericDescPtr dstGenericDescPtr,
+                                       RpptROI3DPtr roiGenericPtrSrc,
+                                       Rpp32f *mulTensor,
+                                       Rpp32f *addTensor,
+                                       rpp::Handle& handle)
+{
+    if (dstGenericDescPtr->layout == RpptLayout::NCDHW)
+    {
+        int globalThreads_x = (dstGenericDescPtr->strides[3] + 7) >> 3; // W - width (x direction) - vectorized for 8 element loads/stores per channel
+        int globalThreads_y = dstGenericDescPtr->dims[3];               // H - height (y direction)
+        int globalThreads_z = dstGenericDescPtr->dims[2];               // D - depth (z direction)
+
+        for(int batchCount = 0; batchCount < dstGenericDescPtr->dims[0]; batchCount++)
```

Comment:
we need to add the multi-stream support into the rpp handle in the future so that we can launch multiple kernels in multiple streams here.